Training/tf/create_dataset.py

```diff
@@ -0,0 +1,126 @@
+            if dataset_cfg['recompute_tau_type']:
+                genLepton_match_map = dict_to_numba(cfg['gen_cfg']['genLepton_match_map'], key_type=types.unicode_type, value_type=types.int32)
+                genLepton_kind_map = dict_to_numba(cfg['gen_cfg']['genLepton_kind_map'], key_type=types.unicode_type, value_type=types.int32)
+                sample_type_map = dict_to_numba(cfg['gen_cfg']['sample_type_map'], key_type=types.unicode_type, value_type=types.int32)
+                tau_type_map = dict_to_numba(tau_type_map, key_type=types.unicode_type, value_type=types.int32)
+                genmatch_dR = compute_genmatch_dR(a)
+                is_dR_matched = genmatch_dR < cfg['gen_cfg']['genmatch_dR']
+
+                tau_type_column = 'tauType_recomputed'
+                a[tau_type_column] = recompute_tau_type(genLepton_match_map, genLepton_kind_map, sample_type_map, tau_type_map,
+                                                            a['sampleType'], is_dR_matched,
+                                                            a['genLepton_index'], a['genJet_index'], a['genLepton_kind'], a['genLepton_vis_pt']) # first execution might be slower due to compilation
+                if sum_:=np.sum(a[tau_type_column]!=a["tauType"]):
+                    print(f'\n        [WARNING] non-zero fraction of recomputed tau types: {sum_/len(a["tauType"])*100:.1f}%\n')
+            else:
+                tau_type_column = 'tauType'
+
+            # create one-hot encoded labels
+            label_columns = []
+            for tau_type, tau_type_value in tau_type_map.items():
+                a[f'label_{tau_type}'] = ak.values_astype(a[tau_type_column] == tau_type_value, np.int32)
+                label_columns.append(f'label_{tau_type}')
+            with open_dict(cfg):
+                cfg["label_columns"] = label_columns
+  
+            time_2 = time.time()
+            print(f'        Preprocessing: took {(time_2-time_1):.1f} s.')
+
+            for tau_type in tau_types:
+                time_2 = time.time()
+                # select only given tau_type
+                a_selected = a[a[f'label_{tau_type}'] == 1]
+                n_selected = len(a_selected)
+                n_samples[tau_type] += n_selected
+                print(f'        Selected: {tau_type}: {n_selected} samples')
+
+                # final tuple with elements to be stored into TF dataset
+                data = ()
+
+                # add awkward arrays converted to TF ragged arrays
+                for particle_type, feature_names in cfg['feature_names'].items(): # do this separately for each particle collection
+                    X = awkward_to_tf(a_selected, particle_type, feature_names) # will keep only feats from feature_names
+                    data += (X,)
+                
+                # all labels to final dataset
+                y = ak.to_pandas(a_selected[cfg["label_columns"]]).values
+                data += (y,)
```

Comment:
I am thinking that storing together tf.array and np.array in one tuple might not be a good way. Also, we might need to check if preliminary converting of X to tensorflow gives better performance than just directly `from_slices((numpy, numpy,))`